### DISCUSSION TITLE
6-to-7-upgrade: I can no longer user get("stage")

### DISCUSSION THREAD

**Discussion by arderyp** [deployphp/deployer#3242] [Help needed] [answered]
I am working on upgrading from 6 to 7 and my custom deployment code is having lots of problems.  The first of which is that `get("stage")` now returns null.  I updated my Host definitions according to the [migration documentation](https://deployer.org/docs/7.x/UPGRADE#step-1-update-deployphp)

Example host:

```
/** @var Localhost $localhost */
$localhost = localhost();
$localhost
    ->setLabels(['stage' => "dev"])
    ->set("branch", "master")
    ->set("http_user", "deployer");
```

Example task:

```
task("host:announce", function () {
    $stage = get("stage"); // RETURNS NULL
    $alias = get("alias");
    info("[$stage] $alias");
})->hidden();
```

For what it is worth, I was using `default_stage` (dev) before, which the migration documentation marked as no longer supported.

**Comment by arderyp** [comment]
Is it safe to also add `set("stage", "dev")` on my host config? That seems to fix the null issue.  I am not sure how/when "labels" come into play.

**Comment by arderyp** [comment]
and, while `set("stage", "dev")` seems to work, it seems deployer still requires a redundant `->setLabels(['stage' => "dev"])` when running `dep stage=dev`. Maybe this is necessary for some reason, or maybe I'm doing something wrong, but it seems redundant to me.

**Comment by arderyp** [comment]
@antonmedv

Thoughts on validity of `set("stage", "dev")`?
Thoughts on redundancy of `set("stage", "dev")` and `->setLabels(['stage' => "dev"])`?

**Reply by antonmedv** [reply]
Didn’t get the question.

**Reply by lerni** [reply]
As far as I understood and also experienced similar, OP tried to get current stage used. Since this returned null he added an extra config.

I used `$stage = Context::get()->getHost()->getConfig()->get('stage');` with 6.x but have not yet found a equivalent for 7.x. With 7 I set `->set('labels', ['stage' => 'test'])` per `host` and would like to get that property in my script.

I do not meant to hickhack the tread but what is a practical way do dump config anyway? I have not set up Xdebug for CLI. Dump it per `writeln`was cutted, cos config is so huge.

**Reply by arderyp** [reply]
@antonmedv @lerni has explained it well. Using `->set('labels', ['stage' => 'dev']` in tandem with `set("stage", "dev")` works for me, but it seems redundant, and like maybe I'm missing a cleaner way to do it with labels alone (since 7 documentation seems to favor labels).

**Reply by antonmedv** [reply]
get(labels)[stage]

**Comment by arderyp** [comment, marked as answer]
as per [this](https://github.com/deployphp/deployer/discussions/3242#discussioncomment-3379723) the solution is `get('labels')['stage']`

EDIT: this should probably be included in the Deployer 7 documentation whenever that gets rolling
